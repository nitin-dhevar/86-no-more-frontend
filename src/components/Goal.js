import React, { Component } from 'react'
import { connect } from 'react-redux'

import { searchUsers } from '../actions/search'
import { editItem } from '../actions/auth'
import { clearsearchstate } from '../actions/search'

import 'react-datepicker/dist/react-datepicker.css'

import { createJob, createInventoryHistory } from '../actions/job'
import { fetchJobs } from '../actions/job'

class Goal extends Component {
    constructor(props) {
        super(props)

        this.state = {
            restname: '',
            restid: '',
            itemname: '',
            quantity: '0',
            costperitem: '',
            datebought: '',
            dateexpired: '',
            editMode: false,
            metric: 'Items',
        }
    }

    handleSave1 = () => {
        const { itemname, quantity, metric } = this.state

        console.log(itemname)

        this.props.dispatch(editItem(itemname, quantity, metric))
        this.props.dispatch(createInventoryHistory(itemname, quantity, metric))
        this.setState({
            itemname: '',
        })

        alert('updated the quantity of ' + itemname)
        document.getElementById('itnameupdate').value = ''
        document.getElementById('quanupdate').value = ''
    }

    clearSearch = () => {
        this.props.dispatch(clearsearchstate([]))
        console.log('UNMOUNT')
    }

    handleInputChange = (fieldName, val) => {
        this.setState({
            [fieldName]: val,
        })
    }

    handleSave = () => {
        const {
            restname,
            restid,
            itemname,
            quantity,
            costperitem,
            datebought,
            dateexpired,
            metric,
        } = this.state

        const { user } = this.props.auth

        this.setState({
            restname: user.restname,
            restid: user._id,
        })

        this.props.dispatch(
            createJob(
                user.restname,
                user._id,
                itemname,
                quantity,
                costperitem,
                datebought,
                dateexpired,
                metric
            )
        )
        this.props.dispatch(createInventoryHistory(itemname, quantity, metric))
        this.setState({
            itemname: '',
        })

        alert(itemname + ' added to the inventory!')
        document.getElementById('itname').value = ''
        document.getElementById('quan').value = ''
        document.getElementById('cost').value = ''
        document.getElementById('edate').value = ''
        document.getElementById('bdate').value = ''
    }

    componentDidMount() {
        this.props.dispatch(fetchJobs())
    }

    render() {
        const { error } = this.props.auth
        const { user } = this.props.auth
        const { job } = this.props

        return (
            <div>
                <div
                    className="goal-form"
                    style={{
                        width: '600px',
                        height: '500px',
                        marginLeft: '100px',
                    }}
                >
                    <span className="login-signup-header">Add Inventory</span>
                    {error && <div className="alert error-dailog">{error}</div>}

                    <div className="field">
                        <input
                            id="itname"
                            placeholder="Item Name"
                            type="text"
                            required
                            onChange={(e) =>
                                this.handleInputChange(
                                    'itemname',
                                    e.target.value
                                )
                            }
                        />
                    </div>

                    <div className="field" style={{ display: 'flex' }}>
                        <input
                            id="quan"
                            placeholder="Quantity"
                            type="text"
                            required
                            onChange={(e) =>
                                this.handleInputChange(
                                    'quantity',
                                    e.target.value
                                )
                            }
                        />
                        <select
                            id="metric"
                            name="selected"
                            defaultValue={'Items'}
                            style={{
                                border: '1px solid rgba(0, 0, 0, 0.12)',
                                height: '40px',
                                marginTop: '20px',
                                padding: '5px',
                                borderRadius: '6px',
                                fontSize: '15px',
                            }}
                            onChange={(e) =>
                                this.handleInputChange('metric', e.target.value)
                            }
                        >
                            <option value="Items">Items</option>
                            <option value="Tons">Tons</option>
                            <option value="Gallons">Gallons</option>
                            <option value="KiloGrams">KiloGrams</option>
                        </select>
                    </div>

                    <div className="field">
                        <input
                            id="cost"
                            placeholder="Cost per Unit"
                            type="text"
                            required
                            onChange={(e) =>
                                this.handleInputChange(
                                    'costperunit',
                                    e.target.value
                                )
                            }
                        />
                    </div>
                    <div className="field">
                        <input
                            id="bdate"
                            placeholder="Date Bought (mm/dd/yyyy)"
                            type="text"
                            required
                            onChange={(e) =>
                                this.handleInputChange(
                                    'datebought',
                                    e.target.value
                                )
                            }
                        />
                    </div>

                    <div className="field">
                        <input
                            id="edate"
                            placeholder="Expiration Date (mm/dd/yyyy)"
                            type="text"
                            required
                            onChange={(e) =>
                                this.handleInputChange(
                                    'dateexpired',
                                    e.target.value
                                )
                            }
                        />
                    </div>

                    <div className="field">
                        <button
                            className="button save-btn"
                            onClick={this.handleSave}
                        >
                            Save
                        </button>
                    </div>
                </div>

                <div
                    className="goal-form"
                    style={{
                        width: '600px',
                        height: '300px',
                        marginLeft: '100px',
                    }}
                >
                    <span className="login-signup-header">Update Item</span>
                    {error && <div className="alert error-dailog">{error}</div>}

                    <div className="field">
                        <input
                            id="itnameupdate"
                            placeholder="Item Name"
                            type="text"
                            required
                            onChange={(e) =>
                                this.handleInputChange(
                                    'itemname',
                                    e.target.value
                                )
                            }
                        />
                    </div>

                    <div className="field" style={{ display: 'flex' }}>
                        <input
                            id="quanupdate"
                            placeholder="Quantity"
                            type="text"
                            required
                            onChange={(e) =>
                                this.handleInputChange(
                                    'quantity',
                                    e.target.value
                                )
                            }
                        />

                        <select
                            id="metric"
                            name="selected"
                            defaultValue={'Tons'}
                            style={{
                                border: '1px solid rgba(0, 0, 0, 0.12)',
                                height: '40px',
                                marginTop: '20px',
                                padding: '5px',
                                borderRadius: '6px',
                                fontSize: '15px',
                            }}
                            onChange={(e) =>
                                this.handleInputChange('metric', e.target.value)
                            }
                        >
                            <option value="Tons">Tons</option>
                            <option value="Gallons">Gallons</option>
                            <option value="KiloGrams">KiloGrams</option>
                        </select>
                    </div>

                    <div className="field">
                        <button
                            className="button save-btn"
                            onClick={this.handleSave1}
                        >
                            Save
                        </button>
                    </div>
                </div>
            </div>
        )
    }
}

function mapStateToProps(state) {
    return {
        auth: state.auth,
        results: state.search.results,
        job: state.job,
    }
}

export default connect(mapStateToProps)(Goal)
