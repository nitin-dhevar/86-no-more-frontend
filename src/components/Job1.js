import React, { Component, useState } from 'react'
import { connect } from 'react-redux'
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
import { faTrash } from '@fortawesome/free-solid-svg-icons'
import { deleteMenu, fetchMenus } from '../actions/job'

class Job1 extends Component {
    constructor(props) {
        super(props)
        this.state = {
            Total: '0',
        }
    }

    handleInputChange = (fieldName, val) => {
        this.setState({
            [fieldName]: val,
        })
    }
    deleteItem = (menuname) => {
        this.props.dispatch(deleteMenu(menuname))
        alert('Are sure, you want to delete ' + menuname + ' from your menu?')
        this.props.dispatch(fetchMenus())
    }

    render() {
        const { job } = this.props
        const { menu } = this.props
        const { user } = this.props.auth
        const { isLoggedIn } = this.props.auth
        this.props.handleCartTotal(this.state.Total * menu.costmenu)
        return (
            <div
                className="post"
                key={menu.restname}
                style={{ width: '50vw', marginLeft: '50px' }}
            >
                {1 > 0 ? (
                    <div className="post-header">
                        <div>
                            <h4 style={{ display: 'inline-block' }}>
                                Restaurant Name :{' '}
                            </h4>
                            <span style={{ marginLeft: '10px' }}>
                                {menu.restname}
                            </span>
                            <button
                                style={{ marginLeft: '300px' }}
                                className="button delete-btn"
                                onClick={() => this.deleteItem(menu.menuname)}
                            >
                                Delete &nbsp;
                                <FontAwesomeIcon
                                    icon={faTrash}
                                    style={{ color: '#f9fafa' }}
                                />
                            </button>
                        </div>

                        <div>
                            <h4
                                style={{
                                    display: 'inline-block',
                                    marginTop: '-12px',
                                }}
                            >
                                Dish :{' '}
                            </h4>
                            <span style={{ marginLeft: '10px' }}>
                                {menu.menuname}
                            </span>
                            <select
                                style={{
                                    marginLeft: '455px',
                                    width: '56px',
                                    height: '37px',
                                    padding: '5px',
                                    border: '2px solid #0000ee',
                                    borderRadius: '6px',
                                    fontSize: '15px',
                                }}
                                onChange={(e) =>
                                    this.handleInputChange(
                                        'Total',
                                        e.target.value
                                    )
                                }
                            >
                                <option value={0}>0</option>
                                <option value={1}>1</option>
                                <option value={2}>2</option>
                                <option value={3}>3</option>
                            </select>
                        </div>

                        <div>
                            <h4
                                style={{
                                    display: 'inline-block',
                                    marginTop: '-12px',
                                }}
                            >
                                Price :{' '}
                            </h4>
                            <span style={{ marginLeft: '10px' }}>
                                {menu.costmenu}
                            </span>
                        </div>

                        <div>
                            <h4
                                style={{
                                    display: 'inline-block',
                                    marginTop: '-12px',
                                }}
                            >
                                Quantity :{' '}
                            </h4>
                            <span style={{ marginLeft: '10px' }}>
                                {menu.quantity}
                            </span>
                        </div>
                        
                        <div>
                            <h4 style={{ display: 'inline-block', marginTop: '-12px' }}>
                                Product Types: 
                            </h4>
                            <div style={{ marginLeft: '10px', display: 'flex', flexWrap: 'wrap' }}>
                                {menu.productTypes && menu.productTypes.map((type, index) => (
                                    <span key={index} style={{ marginRight: '8px', marginBottom: '8px', padding: '6px', border: '1px solid #ddd', borderRadius: '4px', backgroundColor: '#f9f9f9', display: 'inline-block' }}>
                                        {type}
                                    </span>
                                ))}
                            </div>
                        </div>
                        <div>
                            Total Price: {this.state.Total * menu.costmenu}
                        </div>
                    </div>
                ) : (
                    []
                )}
            </div>
        )
    }
}

function mapStateToProps({ auth, job, application }) {
    return {
        auth,
        application,
    }
}

export default connect(mapStateToProps)(Job1)
