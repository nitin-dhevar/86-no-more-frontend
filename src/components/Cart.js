import React, { Component } from 'react'
import { connect } from 'react-redux'
import Job1 from './Job1'
import { fetchJobs, createMenu } from '../actions/job'
import { clearsearchstate } from '../actions/search'
class Cart extends Component {
    constructor(props) {
        super(props)

        this.state = {
            restname: '',
            restid: '',
            menuname: '',
            quantity: '0',
            costmenu: '',
            cartTotal: 0,
            setInput: '0',
            cartChange: '0',
            editMode: false,
        }
    }

    handleInputChange = (fieldName, val) => {
        this.setState({
            [fieldName]: val,
        })
    }

    handleCartTotal = (val) => {
        this.setState({
            ['cartTotal']: this.state.cartTotal + val,
        })
    }

    componentDidMount() {
        this.props.dispatch(fetchJobs())
    }
    clearSearch = () => {
        this.props.dispatch(clearsearchstate([]))
        console.log('UNMOUNT')
    }

    render() {
        const { menu } = this.props
        {console.log('eeeeeee', menu)}
        return (
            <div style={{ display: 'flex' }}>
                <h1>Menu and Cart</h1>
                <div style={{ marginLeft: '57px' }}>
                    {menu?.map((menu) => (
                        <Job1
                            menu={menu}
                            handleCartTotal={this.handleCartTotal}
                        />
                    ))}
                </div>
                <div>
                    <h1 style={{ marginLeft: '100px' }}>
                        {' '}
                        Total: {this.state.cartTotal}
                    </h1>
                    <div>
                        <button
                            style={{ marginLeft: '100px' }}
                            className="button delete-btn"
                            onClick={() => {
                                this.handleInputChange('setInput', '1')
                            }}
                        >
                            Pay
                        </button>
                        <div style={{ marginTop: '20px', marginLeft: '100px' }}>
                            {this.state.setInput === '1' ? (
                                <div>
                                    <input
                                        placeholder="Amount"
                                        style={{
                                            border: '1px solid rgba(0, 0, 0, 0.12)',
                                            height: '40px',
                                            marginTop: '20px',
                                            padding: '5px',
                                            borderRadius: '6px',
                                            fontSize: '15px',
                                        }}
                                        onChange={(e) => {
                                            this.handleInputChange(
                                                'cartChange',
                                                e.target.value
                                            )
                                        }}
                                    ></input>
                                    <h3>
                                        Get change :{' '}
                                        {this.state.cartChange -
                                            this.state.cartTotal}
                                    </h3>
                                </div>
                            ) : null}
                        </div>
                    </div>
                </div>
            </div>
        )
    }
}
function mapStateToProps(state) {
    return {
        auth: state.auth,
        results: state.search.results,
        job: state.job,
        menu: state.menu,
    }
}
export default connect(mapStateToProps)(Cart)
