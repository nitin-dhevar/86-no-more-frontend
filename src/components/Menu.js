import React, { Component } from 'react'
import { connect } from 'react-redux'
import Select from 'react-select'; 
import { searchUsers } from '../actions/search'
import { clearAuthState, editItem } from '../actions/auth'
import { clearsearchstate } from '../actions/search'

import Widgets from './Widgets.js'
import { createJob } from '../actions/job'
import { fetchJobs, createMenu } from '../actions/job'
import Job1 from './Job1'

class Menu extends Component {
    constructor(props) {
        super(props)

        this.state = {
            restname: '',
            restid: '',
            menuname: '',
            quantity: '0',
            costmenu: '',
            selectedProductTypes: [],
            editMode: false,
        }
    }

    handleSearch = (e) => {
        const searchText = e.target.value
        console.log(searchText)

        this.props.dispatch(searchUsers(searchText))
    }
    handleSave1 = () => {
        const {
            restname,
            restid,
            itemname,
            quantity,
            costperitem,
            datebought,
            dateexpired,
        } = this.state

        const { user } = this.props.auth
        const { job } = this.props
        console.log(itemname)

        this.props.dispatch(editItem(itemname, quantity))
    }

    clearSearch = () => {
        this.props.dispatch(clearsearchstate([]))
        console.log('UNMOUNT')
    }

    handleInputChange = (fieldName, val) => {
        this.setState({
            [fieldName]: val,
        })
    }

    handleProductTypeChange = (selectedOptions) => {
        this.setState({
            selectedProductTypes: selectedOptions.map(option => option.value),
        });
        console.log(this.selectedProductTypes);
    }

    handleSave = () => {
        const { restname, restid, menuname, quantity, costmenu, selectedProductTypes } = this.state

        const { user } = this.props.auth

        this.setState({
            restname: user.restname,
            restid: user._id,
        })

        this.props.dispatch(
            createMenu(user.restname, user._id, menuname, quantity, costmenu, selectedProductTypes)
        )
        alert(menuname + ' added!')
    }
    componentDidMount() {
        this.props.dispatch(fetchJobs())
    }

    render() {
        const { error } = this.props.auth

        const { menu } = this.props

        return (
            <div>
                <div
                    className="goal-form"
                    style={{
                        width: '650px',
                        height: '300px',
                        marginLeft: '100px',
                    }}
                >
                    <span className="login-signup-header">Add Menu Item</span>
                    {error && <div className="alert error-dailog">{error}</div>}

                    <div className="field">
                        <input
                            placeholder="Item Name"
                            type="text"
                            required
                            onChange={(e) =>
                                this.handleInputChange(
                                    'menuname',
                                    e.target.value
                                )
                            }
                        />
                    </div>

                    <div className="field">
                        <input
                            id="quan"
                            placeholder="Quantity"
                            type="text"
                            required
                            onChange={(e) =>
                                this.handleInputChange(
                                    'quantity',
                                    e.target.value
                                )
                            }
                        />
                    </div>

                    <div className="field">
                        <input
                            id="cost"
                            placeholder="Price"
                            type="text"
                            required
                            onChange={(e) =>
                                this.handleInputChange(
                                    'costmenu',
                                    e.target.value
                                )
                            }
                        />
                    </div>

                    <div className="field">
                    <label>Select Product Types:</label>
                    <Select
                        isMulti
                        options={[
                            { value: 'Beef', label: 'Beef' },
                            { value: 'Pork', label: 'Pork' },
                            { value: 'Chicken', label: 'Chicken' },
                            { value: 'Milk', label: 'Milk' },
                            { value: 'Egg', label: 'Egg' },
                            { value: 'Vegan', label: 'Vegan' },
                            { value: 'Vegetarian', label: 'Vegetarian' },
                            { value: 'Glutten-Free', label: 'Glutten-Free' },
                            { value: 'Fish', label: 'Fish' },
                            { value: 'Others', label: 'Others'}
                        ]}
                        onChange={this.handleProductTypeChange}
                    />
                </div>

                    <div className="field">
                        <button
                            className="button save-btn"
                            onClick={this.handleSave}
                        >
                            Save
                        </button>
                    </div>
                </div>
            </div>
        )
    }
}

function mapStateToProps(state) {
    return {
        auth: state.auth,
        results: state.search.results,
        job: state.job,
        menu: state.menu,
    }
}

export default connect(mapStateToProps)(Menu)
