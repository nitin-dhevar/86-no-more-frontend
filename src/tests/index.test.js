import React from 'react';
import { mount } from 'enzyme';
import { Provider } from 'react-redux';
import Home from '../components/Home';
import Page404 from '../components/Page404';
import Signup from '../components/Signup';
import Settings from '../components/Settings';
import Goal from '../components/Goal';
import History from '../components/History';
import Update from '../components/Update';
import Notification from '../components/Notification';
import Menu from '../components/Menu';
import Cart from '../components/Cart';
import Job1 from '../components/Job1';
import { configureStore } from '../store';

const store = configureStore();

test('renders Home component', () => {
  const wrapper = mount(
    <Provider store={store}>
      <Home />
    </Provider>
  );

  expect(wrapper).toMatchSnapshot();
});

// Page404.test.js
test('renders Page404 component', () => {
  const wrapper = mount(
    <Provider store={store}>
      <Page404 />
    </Provider>
  );

  expect(wrapper).toMatchSnapshot();
});
test('renders Signup component', () => {
    const wrapper = mount(
      <Provider store={store}>
        <Signup />
      </Provider>
    );
  
    expect(wrapper).toMatchSnapshot();
  });
  test('renders Settings component', () => {
    const wrapper = mount(
      <Provider store={store}>
        <Settings />
      </Provider>
    );
  
    expect(wrapper).toMatchSnapshot();
  });
  test('renders Goal component', () => {
    const wrapper = mount(
      <Provider store={store}>
        <Goal />
      </Provider>
    );
  
    expect(wrapper).toMatchSnapshot();
  });
  test('renders History component', () => {
    const wrapper = mount(
      <Provider store={store}>
        <History />
      </Provider>
    );
  
    expect(wrapper).toMatchSnapshot();
  });
  test('renders Update component', () => {
    const wrapper = mount(
      <Provider store={store}>
        <Update />
      </Provider>
    );
  
    expect(wrapper).toMatchSnapshot();
  });
  test('renders Notification component', () => {
    const wrapper = mount(
      <Provider store={store}>
        <Notification />
      </Provider>
    );
  
    expect(wrapper).toMatchSnapshot();
  });
  test('renders Menu component', () => {
    const wrapper = mount(
      <Provider store={store}>
        <Menu />
      </Provider>
    );
  
    expect(wrapper).toMatchSnapshot();
  });
  test('renders Cart component', () => {
    const wrapper = mount(
      <Provider store={store}>
        <Cart />
      </Provider>
    );
  
    expect(wrapper).toMatchSnapshot();
  });
  