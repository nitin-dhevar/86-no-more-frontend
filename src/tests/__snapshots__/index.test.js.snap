// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Cart component 1`] = `ReactWrapper {}`;

exports[`renders Goal component 1`] = `ReactWrapper {}`;

exports[`renders History component 1`] = `ReactWrapper {}`;

exports[`renders Home component 1`] = `ReactWrapper {}`;

exports[`renders Menu component 1`] = `ReactWrapper {}`;

exports[`renders Notification component 1`] = `ReactWrapper {}`;

exports[`renders Page404 component 1`] = `ReactWrapper {}`;

exports[`renders Settings component 1`] = `ReactWrapper {}`;

exports[`renders Signup component 1`] = `ReactWrapper {}`;

exports[`renders Update component 1`] = `ReactWrapper {}`;
